packages/react-async/documentation/migrations.md

```diff
@@ -0,0 +1,96 @@
+# Migration guides
+
+## 1.x to 2.x
+
+The 2.x branch of `@shopify/react-async` is a significant update that brings progressive hydration, more flexible prefetching, the ability to preload assets in the server render, and more. These changes required some significant changes to the library.
+
+### `renderX` options become `useX`
+
+Flexible preloading/ prefetching/ keeping fresh depend on having those operations be implemented as hooks, rather than being implemented as components. Most examples of using using this library pre-2.x would involve rendering components to preload, like so:
+
+```tsx
+import {createAsyncComponent} from '@shopify/react-async';
+import {MyComponentsQuery} from './components';
+
+const MyComponent = createAsyncComponent({
+  load: () => import('./MyComponent'),
+  renderPreload: () => <MyComponentsQuery.Preload />,
+  renderPrefetch: () => <MyComponentsQuery.Prefetch />,
+  renderKeepFresh: () => <MyComponentsQuery.KeepFresh />,
+});
+```
+
+This was particularly common when using `@shopify/react-graphql`, which was designed to work well in conjunction with the `renderX` methods.
+
+In the 2.x branch, we no longer support rendering components, only using hooks. This should be a strictly more powerful technique, as it enables you to do some work up front, or delay the work into a callback that is returned. Updating will depend on exactly what was rendered through the `renderX` options. If you were only using it for tying the async component to an async query, you can use the following changes, assuming you are on the 4.x branch of `@shopify/react-graphql`:
+
+```diff
+import {
+  createAsyncComponent,
++  usePreload,
++  usePrefetch,
++  useKeepFresh,
+} from '@shopify/react-async';
+import {MyComponentsQuery} from './components';
+
+const MyComponent = createAsyncComponent({
+  load: () => import('./MyComponent'),
+-  renderPreload: () => <MyComponentsQuery.Preload />,
+-  renderPrefetch: () => <MyComponentsQuery.Prefetch />,
+-  renderKeepFresh: () => <MyComponentsQuery.KeepFresh />,
++  usePreload: () => usePreload(MyComponentsQuery),
```

Comment:
O.o this api is super weird. I feel like the `use` prefix is confusing here since these are properties not hooks?

I'd maybe drop the use prefix and have one property you can pass a component to that will automatically call the corresponding hook for each?

Reply:
They are hooks, though. Whatever gets called here gets called as part of the final useX, and must obey the rules of hooks.